### PR TITLE
Automate nox version upgrade

### PR DESCRIPTION
**Pull request by seanblumenfeld**
This change introduces a Github action to create a pull request whenever a new version of nox is detected on PyPi.

 The Github action does the following:
  - installs the latest version of nox from PyPi
  - spits out that version into a requirements.txt file
  - uses peter-evans/create-pull-request@v3 github action to create a pull request if any files have changed. Thus, if a new version of nox has been found on PyPi the requirements.txt file will be detected as a changed file and a pull request will be created with the updated nox version.

 If desired the settings for the pull request creation can be changed to auto merge. I'll leave that decision up to you @TheKevJames

### CODE REVIEW

**Comment by seanblumenfeld** [comment]
Related to issue https://github.com/TheKevJames/tools/issues/2

**Comment by seanblumenfeld** [comment]
I just noticed that you are already using renovate in this repo. So I have changed my PR to use a simple renovate config to manage the nox version.

File: .github/renovate.json [hunk 1]
```diff
@@ -8,5 +8,10 @@
     "groupName": "phoenix_view",
     "managers": ["mix"],
     "separateMajorMinor": false
-  }]
+  }],
+  "pip_requirements": {
```
Comment (TheKevJames) [inline .github/renovate.json:12]: Shouldn't be any need for changes to this file, since renovate should run on all `requirements.txt` files by default.

**Comment by TheKevJames** [comment]
Tackled this [here](https://github.com/TheKevJames/tools/commit/87ad6d872edde6ecb04c160725ba9688d398a7bb) and [here](https://github.com/TheKevJames/tools/commit/4dc3de14c0c103e29cfc36674527fb38229d237f). Thanks for the starting point! Should now do the autobuild/merge/tag/deploy stuff as well.

**Pull request closed without merging**